utility/src/main/java/org/oppia/android/util/parser/html/HtmlParser.kt

```diff
@@ -75,27 +91,38 @@ class HtmlParser private constructor(
       htmlContent = htmlContent.replace("\n\n", "")
     }
     if ("<li>" in htmlContent) {
-      htmlContent = htmlContent.replace("<li>", "<$CUSTOM_BULLET_LIST_TAG>")
-        .replace("</li>", "</$CUSTOM_BULLET_LIST_TAG>")
+      htmlContent = htmlContent.replace("<li>", "<$CUSTOM_LIST_LI_TAG>")
+        .replace("</li>", "</$CUSTOM_LIST_LI_TAG>")
+    }
+    if ("<ul>" in htmlContent) {
+      htmlContent = htmlContent.replace("<ul>", "<$CUSTOM_LIST_UL_TAG>")
+        .replace("</ul>", "</$CUSTOM_LIST_UL_TAG>")
+    }
+    if ("<ol>" in htmlContent) {
+      htmlContent = htmlContent.replace("<ol>", "<$CUSTOM_LIST_OL_TAG>")
+        .replace("</ol>", "</$CUSTOM_LIST_OL_TAG>")
     }
 
     // https://stackoverflow.com/a/8662457
     if (supportsLinks) {
       htmlContentTextView.movementMethod = LinkMovementMethod.getInstance()
+      LinkifyCompat.addLinks(htmlContentTextView, Linkify.WEB_URLS)
+    }
+
+    val imageGetter: UrlImageParser?
+    if (urlImageParserFactory == null) {
+      imageGetter = null
+    } else {
+      imageGetter = urlImageParserFactory?.create(
+        htmlContentTextView, gcsResourceName, entityType, entityId, imageCenterAlign
+      )
     }
```

Comment:
```suggestion
    val imageGetter = urlImageParserFactory?.create(
      htmlContentTextView, gcsResourceName, entityType, entityId, imageCenterAlign
    )
```

Equivalent & more Kotlin-idiomatic.